### ISSUE TITLE
Can it be controlled or control other RGB applications?

### ISSUE THREAD

**Issue by SmellsSmell** [OmenMon/OmenMon#47]
Hi, I have been looking for an alternative to my OMEN Gaming Hub for a long time and only recently stumbled across your amazing open-source software for the second time recently. Last time I was told about it by a family member and it wasn't yet able to run my laptop. But it's been fixed since then. And props for all the good work. This software feels so much better than the original OGH. 

However, one of the main reasons I want to replace OGH is not because it's bloatware, though it is. I want more RGB control on my keyboard. OMEN Light Studio allows you to use the background application to control other devices. But it refuses to pick up on other devices that I have. So is there any possible way I can get OMENMon to be controlled by other software?

I am on a **Victus 16-r0073cl with a core i7 13700hx and a mobile RTX 4060**. My laptop is extremely hard to find on the HP Website, and the only way I have found it is through the link in the myHP app. Even searching through model numbers and using my S/N wouldn't give me anything. And, it's not even listed as a supported product by OMEN Light Studio. So maybe if I could be pointed in the direction of another Open-Source application that allows me to control the lights through the WMI bios and does the connect feature. But considering not a lot of people are trying to do this kind of stuff, I haven't been able to find anything online about something like that. 

On your website, in the features section under application, it says "Does exactly what you expect it to do and nothing else." So, I am curious, is this functionality perhaps already built in, and because of the model of laptop I am using I can't use it? Because like I said OLS technically doesn't list my device as supported. So maybe I'd have to look into getting something made from scratch if I want to do this. It would be nice if the functionality was built into OMENMon though because then I can just replace OGH and be done.

TL;DR
I have a Victus Laptop with OGH and I would like to be able to sync the keyboard's 1 zone RGB with other software i.e. Razer or Aurora. Victus laptops are not currently listed as supported under OMEN Light Studio, so maybe this has to be done differently than I thought, you've got the experience with it to be able to find that out. No rush though, just a minor cosmetic thing. And the software as it is right now is pretty amazing.

**Comment by GeographicCone** [comment]
**OmenMon**'s backlight support is pretty basic, and it doesn't work with the per-key RGB models.

That being said, if it does work for you, it is accessible from the CLI mode with `OmenMon -Bios Color=...`, which also means it can be automated and controlled from another application. [See here.](https://omenmon.github.io/cli#color)

For many other devices, there is the [OpenRGB](https://openrgb.org/) project. I haven't had the chance to try it myself but feature-wise it seems quite promising.

If color settings work for you from the hardware side, and it's only about making them dynamic, it's currently not possible with the GUI but should be doable with a script calling the CLI repeatedly in a loop. There is some overhead to doing it is way but I doubt it'd be noticeable on a powerful system (that already has a lot of more egregious garbage running in the background anyway, which is beyond user control). In the long run, such a feature would be interesting to have integrated into **OmenMon**.

As for your model not being listed or searchable on the website, I think it's par for the course. There are many models that only differ by the keyboard layout, and they never bothered listing most of them, only the US equivalent. What's surprising though is that your model seems to be from the US market, so shouldn't be affected by this unofficial long-standing "policy." Well, maybe now they care about it even less than before. Go figure. Anyway, _Victus_ and _Omen_ seem to be very similar internally, so most of the Omen stuff should work just as well.
